src/hotspot/share/oops/symbolHandle.hpp

```diff
@@ -84,8 +111,22 @@ class SymbolHandleBase : public StackObj {
   static unsigned int compute_hash(const SymbolHandleBase& name) {
     return (unsigned int) name->identity_hash();
   }
+
+  static void drain_cleanup_delay_queue() {
```

Comment:
It's not obvious that draining the queue is useful. Unless there's a reason I'm missing, I suggest not doing so.

Reply:
I didn't find any discussion of whether draining is needed in this PR, and draining is in the initial commit.
Other downsides include the need to test that feature and the impact that feature has on testing other parts
of this change.  Unless someone argues for it, I'd prefer to see it removed.

Reply:
Yes, there is one additional test for the draining but it's quite simple. The drain feature actually helps with the other tests, that's how we can avoid the queue interfering with ref counts for the tests (create temp symbol, immediately drain queue), so even if we don't drain periodically I'd still leave the feature there for the existing tests.

What about the fact that not draining the queue means we could keep 128 symbols alive indefinitely? Are you not concerned because that's a small amount, and/or because natural churn will likely keep the queue moving?

Reply:
I think the footprint savings is probably negligible.  I like the idea that the queue is periodically drained because it might show issues where we've messed up the refcounts, but they'd be hard to debug, so maybe not worth it. So I see neither the harm nor the benefit of draining the queue in the periodic task, but we should have the feature for the one test.

Reply:
I've removed drain from the concurrent cleanup. I left it public and tested, is that reasonable or do you prefer I make it private somehow (not sure the typical way to have test-only methods)?

Reply:
I like that there's a test.  Some gtests use friend declarations if they're in a class, but test_symbolTable isn't. I don't think it's worth changing.